ckanext/datasetapproval/helpers.py

```diff
@@ -16,3 +19,54 @@ def is_dataset_owner(data_dict, user_id):
         return dataset["creator_user_id"] == user_id
     except Exception as e:
         return False
+
+
+def get_dataset_user_permissions(dataset_dict, user_obj):
```

Comment:
I centralized the permissions control based on the dataset labels and user labels in this function so that it's easy to update it.